File: CO322-Data Structures and Algorithms/Lab1- run time of an algorithm/fib.py
#!/usr/bin/python

def fib_r(x) :
    if (x <= 2):
        return 1
    return fib_r(x-1) + fib_r(x-2)


def fib_i(x) :
    a = 1
    b = 1
    fib = 1
    i = 2
    while i < x:
        fib = a +b
        a = b
        b = fib
        i+=1
        
    return fib

x = 40

for i in range (0,41):
    import time
    start = time.time()
    k = str(fib_r(i))
    #print(k)
    #print("Fib of " + str(i) + " = " + str(fib_r(i)))
    endR = time.time()
    print((endR - start)*1000)

print("---------------------")    

for j in range (0,41):
    import time
    start = time.time()
    l = str(fib_i(j))
    #print(l)
    #print ("Fib of " + str(j) + " = " + str(fib_i(j)))
    endI = time.time()
    print((endI - start)*1000)

File: CO322-Data Structures and Algorithms/Lab1- run time of an algorithm/Fib.java
class Fib {

    public static int fib_r(int x) {
	if(x <= 2) return 1;
	return fib_r(x-1) + fib_r(x-2);
    }

    public static int fib_i(int x) {
	int a=1, b=1, fib=1, i=2;
	while(i<x) {
	    fib = a + b;
	    a = b;
	    b = fib;
	    i+=1;
	}
	return fib;
    }


    public static void main(String [] args) {
	int x = 40;

  for (int j=0;j<=x ;j++ ) {
    long startTime = System.nanoTime();
    long fibr = fib_r(j);
    // System.out.println("Fib of " + j + " = " + fib_r(j));
    long endTime   = System.nanoTime();
    long totalTime = endTime - startTime;
    System.out.print((totalTime)/1000000+",");
  }

System.out.println(".............................................................");

  for (int j=0;j<=x ;j++ ) {
    long startTime_i = System.nanoTime();
    long fibi = fib_i(j);
    // System.out.println("Fib of " + j + " = " + fib_r(j));
    long endTime_i   = System.nanoTime();
    long totalTime_i = endTime_i - startTime_i;
    System.out.print((totalTime_i)/1000000+",");
  }

    }
}
